src/parser/parser_nodes.nit

```diff
@@ -311,6 +311,33 @@ abstract class Token
 	# May have disappeared in the AST
 	var next_token: nullable Token = null
 
+	# Is `self` a token discarded from the AST?
+	#
+	# Loose tokens are not present in the AST.
+	# It means they are parsed by the lexer but where discarded by the parser.
```

Comment:
Actually, your sentence is both in the present and passive form, it bothers me.

Maybe use `but are discarded`